pkg/graveler/graveler.go

```diff
@@ -471,6 +471,13 @@ type CommitParams struct {
 	AllowEmpty      bool
 }
 
+type CommitOverrides struct {
```

Comment:
Please add documentation on this struct as it's not very intuitive (repeats some of the Commit parameters but behaves differently)

Reply:
Also, here's a little something you can do with golang to avoid duplication (not sure you'd want to here and if you do, find a better name than `Base`):
```golang
type CommitParamsBase struct {
	Message   string
	// Date (Unix Epoch in seconds) is used to override commits creation date
	Date     *int64
	Metadata Metadata
}

type CommitParams struct {
        CommitParamsBase
	Committer string
	// SourceMetaRange - If exists, use it directly. Fail if branch has uncommitted changes
	SourceMetaRange *MetaRangeID
	AllowEmpty      bool
}

```